packages/default/scss/drawer/_layout.scss

```diff
@@ -1,155 +1,164 @@
-                margin: 0 $drawer-item-padding-x;
-            }
+    // Overlay drawer
+    .k-drawer-overlay {
 
-            .k-item-text {
-                display: inline-block;
-                padding: 0 $drawer-item-padding-x;
-                vertical-align: middle;
-            }
+        .k-drawer {
+            max-width: 80vw; // limit width
+            position: fixed;
+            z-index: 10002;
         }
 
-        &.k-drawer-expanded {
-            border-right-width: $drawer-border-width;
+        .k-overlay {
+            display: none;
         }
+        &.k-drawer-expanded .k-overlay {
+            display: block;
+        }
+
     }
 
-    .k-drawer-wrapper {
-        overflow-x: hidden;
 
-        &.k-drawer-mini-mode {
-            .k-drawer-container {
-                border-right-width: $drawer-border-width;
-                width: auto;
-            }
+    // Push drawer
+    .k-drawer-push {
+        display: flex;
+        flex-direction: row;
 
-            .k-drawer-items {
-                overflow-y: visible;
-                overflow-x: initial;
-                width: $drawer-mini-initial-width; // Default width of mini sidebar
-            }
+        .k-drawer {
+            height: auto;
+            position: relative;
+            align-self: stretch;
         }
 
-        &.k-drawer-overlay {
-            .k-drawer-container {
-                position: fixed;
-                left: 0;
-                right: auto;
-                height: 100%;
-                z-index: 10002;
-            }
 
-            .k-drawer-items {
-                max-width: 80vw;
+        // right drawer
+        &.k-drawer-right {
+            .k-drawer {
+                order: 1;
             }
         }
 
+    }
 
-        &.k-drawer-push {
-            display: flex;
-            height: 100%;
 
-            .k-drawer-content {
-                flex: 1 1 auto;
-            }
+    // Drawer items
+    .k-drawer-wrapper {
+        overflow-x: hidden;
+        overflow-y: auto;
+        scrollbar-width: thin; // Scrollbar styles for Mozilla
+        transition: all 300ms ease-in-out; // sass-lint:disable-line no-transition-all
 
-            .k-drawer-container {
-                position: relative;
-                flex: 0 0 auto;
-            }
+        // Scrollbar styles for Chrome, Safari and Opera
+        &::-webkit-scrollbar {
+            width: $drawer-scrollbar-width;
+        }
+        &::-webkit-scrollbar-thumb {
+            border-radius: $drawer-scrollbar-radius;
+        }
+    }
+    .k-drawer-items,
+    .k-drawer-items ul {
+        margin: 0;
+        padding: 0;
+        list-style: none;
+        display: flex;
+        flex-direction: column;
+        align-items: stretch;
+    }
+    .k-drawer-item {
+        padding: $drawer-item-padding-y $drawer-item-padding-x;
+        color: inherit;
+        font-size: $drawer-item-font-size;
+        line-height: $drawer-item-line-height;
```

Comment:
The drawer items text and icons look misaligned (or aligned bottom).
![image](https://user-images.githubusercontent.com/8842674/64970355-4977ae00-d8ae-11e9-96ba-ecccba4e1e03.png)